internal/pkg/graph/graph.go

```diff
@@ -0,0 +1,97 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package graph provides functionality for directed graphs.
+package graph
+
+// nodeColor denotes the color of a node when running DFS in a graph.
+type nodeColor int
+
+const (
+	white nodeColor = iota + 1
+	gray
+	black
```

Comment:
```suggestion
	white nodeColor = iota + 1 // A node is colored white if it is unvisited.
	gray      // A node is colored gray if it's visited but isn't finished (hence can potentially lead to a cycle).
	black     // A node is colored black if it's finished (hence not possible to lead to a cycle).
```
Or we could just rename them as `unvisited`, `visited`, and `skip`/`finished`, instead of adding comments 💭 .